bigarray should not be defined in OCaml 5.00

I think dune needs additional fixes for OCaml 5.00
https://github.com/ocaml/dune/blob/2.9/src/dune_rules/findlib/meta.ml#L230
`bigarray` is defined as dummy but has been removed in OCaml 5.00
This causes issues for packages that use ocamlfind:
```
#=== ERROR while compiling topkg-care.1.0.5 ===================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-variants.5.00.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/5.00/.opam-switch/build/topkg-care.1.0.5
# command              ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml build --pkg-name topkg-care --dev-pkg false
# exit-code            1
# env-file             ~/.opam/log/topkg-care-10-c10615.env
# output-file          ~/.opam/log/topkg-care-10-c10615.out
### output ###
# ocamlfind ocamldep -package 'opam-format cmdliner fmt.cli logs.cli bos.setup' -package 'webbrowser webbrowser.cli' -modules src-care/topkg_care.ml > src-care/topkg_care.ml.depends
# + ocamlfind ocamldep -package 'opam-format cmdliner fmt.cli logs.cli bos.setup' -package 'webbrowser webbrowser.cli' -modules src-care/topkg_care.ml > src-care/topkg_care.ml.depends
# ocamlfind: Package `bigarray' not found - required by `opam-core'
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
#      '-build-dir' '_build' 'CHANGES.md' 'LICENSE.md' 'README.md'
#      'topkg-care.opam' 'src-care/topkg_care.a' 'src-care/topkg_care.cmxs'
#      'src-care/topkg_care.cmxa' 'src-care/topkg_care.cma'
#      'src-care/topkg_care_text.cmx' 'src-care/topkg_care_pkg.cmx'
#      'src-care/topkg_care_opam.cmx' 'src-care/topkg_care_ocamlfind.cmx'
#      'src-care/topkg_care_ocamlbuild.cmx' 'src-care/topkg_care_ipc.cmx'
#      'src-care/topkg_care_delegate.cmx' 'src-care/topkg_care_archive.cmx'
#      'src-care/topkg_care.cmx' 'src-care/topkg_care.cmi'
#      'src-care/topkg_care.mli' 'src-bin/topkg_bin.native'
#      'src-bin/toy_github_delegate.native' 'test/unsupportive-delegate'
#      'test/echo-delegate']: exited with 10
```
Where `opam-core` requires `bigarray` but isn’t notified that the library does not exist anymore, thus making any subsequent library fail.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -310,6 +310,12 @@ Unreleased
 - Bootstrap: ignore errors when trying to remove generated files. (#5407,
   @damiendoligez)
 
+2.9.4 (unreleased)
+------------------
+
+- Do not generate META information for `bigarray` library in OCaml >= 5.0
+  (#5421, @nojb)
+
 2.9.3 (unreleased)
 ------------------
 

--- a/src/dune_rules/findlib/meta.ml
+++ b/src/dune_rules/findlib/meta.ml
@@ -279,16 +279,13 @@ let builtins ~stdlib_dir ~version:ocaml_version =
   in
   let libs =
     let base =
-      [ stdlib
-      ; compiler_libs
-      ; str
-      ; unix
-      ; bigarray
-      ; threads
-      ; dynlink
-      ; bytes
-      ; ocamldoc
-      ]
+      [ stdlib; compiler_libs; str; unix; threads; dynlink; bytes; ocamldoc ]
+    in
+    let base =
+      if Ocaml_version.has_bigarray_library ocaml_version then
+        bigarray :: base
+      else
+        base
     in
     let base =
       if Ocaml_version.stdlib_includes_uchar ocaml_version then

--- a/src/dune_rules/ocaml_version.ml
+++ b/src/dune_rules/ocaml_version.ml
@@ -49,3 +49,5 @@ let ocamlopt_always_calls_library_linker version = version < (4, 12, 0)
 let has_sys_opaque_identity version = version >= (4, 3, 0)
 
 let has_vmthreads version = version < (4, 9, 0)
+
+let has_bigarray_library version = version < (5, 0, 0)

--- a/src/dune_rules/ocaml_version.mli
+++ b/src/dune_rules/ocaml_version.mli
@@ -68,3 +68,6 @@ val has_sys_opaque_identity : t -> bool
 
 (** Whether [vmthreads] exists *)
 val has_vmthreads : t -> bool
+
+(** Whether [bigarray] {em library} exists *)
+val has_bigarray_library : t -> bool

